tools/test_crypto.py

```diff
@@ -0,0 +1,156 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import os
+import sys
+import git
+
+from subprocess import check_call
+import argparse
+SCRIPT_DIRECTORY = os.path.dirname(os.path.realpath(__file__)) + "/"
+MBEDTLS_IMPORTER_DIRECTORY = SCRIPT_DIRECTORY + "../features/mbedtls/importer/"
+MBED_CRYPTO_DIRECTORY = MBEDTLS_IMPORTER_DIRECTORY + "/TARGET_IGNORE/mbedtls/crypto/"
+MBED_CRYPTO_TEST_DIRECTORY = MBED_CRYPTO_DIRECTORY + "/tests/"
+MBED_OS_TEST_DIRECTORY = SCRIPT_DIRECTORY + "../TESTS/"
+
+
+logger = logging.getLogger('PSA-CRYPTO Builder')
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="This script build psa-crypto on-target tests."
+    )
+
+    parser.add_argument(
+        '-c',
+        '--clean',
+        action='store_true',
+        help="choose a specific test to compile"
+    )
+
+    parser.add_argument(
+        '-r',
+        '--run',
+        action='store_true',
+        help="Run psa crypto tests"
+    )
+
+    parser.add_argument(
+        '-b',
+        '--build',
+        action='store_true',
+        help="Build psa crypto tests"
+    )
+
+    parser.add_argument(
+        '-g',
+        '--generate',
+        action='store_true',
+        help='Generate and copy psa-crypto tests to mbed-os TESTS folder'
+    )
+
+    parser.add_argument(
+        '-d',
+        '--deploy',
+        action='store_true',
+        help='Deploy mbed-crypto and mbed-tls through importer'
+    )
+    parser.add_argument(
+        '-p',
+        '--profile',
+        choices=['debug', 'developer', 'release'],
+        default='developer',
+        help='mbed-os build profile'
+    )
+    parser.add_argument(
+        '-t',
+        '--toolchain',
+        choices=['GCC_ARM', 'ARM', 'IAR'],
+        default='GCC_ARM',
+        help='mbed-os build toolchain'
+    )
+    parser.add_argument(
+        '-m',
+        '--target',
+        default='K64F',
+        help='mbed-os build target'
+    )
+
+    return parser
+
+
+def psa_deploy():
+    logger.info("--- Set up psa-crypto env ---")
+
+    check_call(["make", "update"], cwd=MBEDTLS_IMPORTER_DIRECTORY)
+    crypto_repo = git.Repo(MBED_CRYPTO_DIRECTORY)
+    crypto_repo.git.checkout("feature-psa")
+    check_call(["make"], cwd=MBEDTLS_IMPORTER_DIRECTORY)
+    logger.info("--- Set up psa-crypto env Done---")
+
+
+def generate_tests():
+    logger.info("--- generate psa-crypto tests start---")
+    check_call(["make", "generate-target-tests"], cwd=MBED_CRYPTO_TEST_DIRECTORY)
+    os.system("cp -r " + MBED_CRYPTO_TEST_DIRECTORY + "/TESTS/*" + " " + MBED_OS_TEST_DIRECTORY)
+    os.system("cp " + MBED_CRYPTO_TEST_DIRECTORY + "/scripts/mbedtls_test.py" + " " +  MBED_OS_TEST_DIRECTORY
```

Comment:
You can remove the depenency on the `cp` utility (missing in the default Windows environment) by using [`shutil.copytree`](https://docs.python.org/2/library/shutil.html#shutil.copytree).

Reply:
Do you need some pointers to implement this one @netanelgonen? It would mean one less dependency on the host environment.